packages/editor/src/components/post-actions/actions.js

```diff
@@ -982,95 +973,96 @@ const renameTemplateAction = {
 	},
 };
 
-export function usePostActions( onActionPerformed, actionIds = null ) {
+export function usePostActions( postType, onActionPerformed ) {
+	const { postTypeObject } = useSelect(
+		( select ) => {
+			const { getPostType } = select( coreStore );
+			return {
+				postTypeObject: getPostType( postType ),
+			};
+		},
+		[ postType ]
+	);
+
 	const permanentlyDeletePostAction = usePermanentlyDeletePostAction();
 	const restorePostAction = useRestorePostAction();
-	return useMemo(
-		() => {
-			// By default, return all actions...
-			const defaultActions = [
-				editPostAction,
-				resetTemplateAction,
-				viewPostAction,
-				restorePostAction,
-				deleteTemplateAction,
-				permanentlyDeletePostAction,
-				postRevisionsAction,
-				duplicatePostAction,
-				renamePostAction,
-				renameTemplateAction,
-				trashPostAction,
-			];
+	const isTemplateOrTemplatePart = [
+		TEMPLATE_POST_TYPE,
+		TEMPLATE_PART_POST_TYPE,
+	].includes( postType );
+	const isPattern = postType === PATTERN_POST_TYPE;
+	const isLoaded = !! postTypeObject;
+	return useMemo( () => {
+		if ( ! isLoaded ) {
+			return [];
+		}
 
-			// ... unless `actionIds` was specified, in which case we find the
-			// actions matching the given IDs.
-			const actions = actionIds
-				? actionIds.map( ( actionId ) =>
-						defaultActions.find( ( { id } ) => actionId === id )
-				  )
-				: defaultActions;
+		const actions = [
+			isTemplateOrTemplatePart && resetTemplateAction,
```

Comment:
I wonder if it would not be better to do the checks for post type on the isEligible function of the action it self.

Reply:
I thought about it, I do think it would be better but I'm not sure that the current API allows us to do that. The isEligible can't call selectors with resolvers at the moment and be certain that the value is there.

Reply:
I'm also not sure 100% that it's better. This is the mental model I have for now:

 - isEligible is for actions that can apply to one item of the current dataviews/list but may not apply for a reason that is specific to the current item.
 - The actions you pass to DataViews (or the Actions component) are all "valid" for the current post type.

Reply:
On the item itself, the rest API provides some checks in wp:action... that allows some possibilities but yes it is very limited.
I imagine a plugin may want to build a view that shows all posts from all types in the database in that case we would want all actions there, and depending on the item some would be available, and some would not.
But given that we are keeping actions as hooks we can keep the current approach the current mental modal long term maybe isEligible should be an async function.